src/stenotype/backend/elements.py

```diff
@@ -0,0 +1,199 @@
+        return True if type(other) is Dots else NotImplemented
+
+    def __str__(self):
+        return "..."
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}()"
+
+
+class Identifier(TupleT[str, ...]):
+    """A qualified name, such as ``typing.Tuple``"""
+
+    def __new__(cls, *elements: str):
+        assert all(type(el) is str and el.isidentifier() for el in elements)
+        return super().__new__(cls, elements)
+
+    def __str__(self):
+        return ".".join(self)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({", ".join(map(repr, self))})'
+
+
+class Generic(NamedTuple):  # type: ignore
+    """A generic type, such as ``typing.List[int]``"""
+
+    base: Identifier
+    parameters: TupleT["Steno", ...]
+
+    def __str__(self):
+        return f'{self.base}[{", ".join(map(str, self.parameters))}]'
+
+
+class Any:
+    """Any type as ``_``, same as :py:class:`typing.Any`"""
+
+    __slots__ = ()
+
+    def __eq__(self, other):
+        return True if type(other) is Any else NotImplemented
+
+    def __str__(self):
+        return "_"
+
+    def __repr__(self):
+        return f"Any()"
+
+
+class Optional(NamedTuple):  # type: ignore
+    """Optional type as ``?base``, equivalent to ``typing.Optional[base]``"""
+
+    base: "Steno"
+
+    def __str__(self):
+        return f"?{self.base}"
+
+
+class Union(TupleT["Steno"]):
+    """Union of types as ``A or B``, equivalent to ``typing.Union[A, B]``"""
+
+    def __new__(cls, *elements: "Steno"):
+        return super().__new__(cls, elements)
+
+    def __str__(self):
+        return " or ".join(map(str, self))
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({", ".join(map(repr, self))})'
+
+
+class Tuple(NamedTuple):  # type: ignore
+    """Typed tuple as ``(A, B, ...)``, equivalent to ``typing.Tuple[A, B, ...]``"""
+
+    elements: TupleT[UnionT["Steno", Dots], ...]
+
+    def __str__(self):
+        return f'({", ".join(map(str, self.elements))})'
+
+
+class List(NamedTuple):  # type: ignore
+    """Typed list as ``[values]``, equivalent to ``typing.List[values]``"""
+
+    values: "Steno"
+
+    def __str__(self):
+        return f"[{self.values}]"
+
+
+class Dict(NamedTuple):  # type: ignore
+    """Typed dict as ``{keys: values}``, equivalent to ``typing.Dict[keys, values]``"""
+
+    keys: "Steno"
+    values: "Steno"
+
+    def __str__(self):
+        return f"{{{self.keys}: {self.values}}}"
+
+
+class Set(NamedTuple):  # type: ignore
+    """Typed dict as ``{values}``, equivalent to ``typing.Set[values]``"""
+
+    values: "Steno"
+
+    def __str__(self):
+        return f"{{{self.values}}}"
+
+
+class Literal(NamedTuple):
+    """Literal value, not a type"""
+
+    value: UnionT[None, bool, int, str, bytes, "ellipsis"]
+
+    def __str__(self):
+        return str(self.value)
+
+
+class Iterable(NamedTuple):  # type: ignore
+    """Typed iterable as ``iter base``, equivalent to ``typing.Iterable[base]"""
```

Comment:
Typo, closing backticks
```suggestion
    """Typed iterable as ``iter base``, equivalent to ``typing.Iterable[base]``"""
```